### PR TITLE
Keep namespace after test run with with RELEASE_NAMESPACE=false

### PR DESCRIPTION
**Pull request by psegedy**
`KEEP_NAMESPACE` is unset or empty string -> release namespace
`KEEP_NAMESPACE` set to anything else -> set it to `TRUE` and keep namespace

it would be useful for debugging failing PR checks

### CODE REVIEW

File: cicd/_common_deploy_logic.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #COMPONENTS="component1 component2"  # specific components to deploy (optional, default: all)
 #COMPONENTS_W_RESOURCES="component1 component2"  # components which should preserve resource settings (optional, default: none)
 #DEPLOY_TIMEOUT="300"  # bonfire deployment timeout parameter in seconds
+#KEEP_NAMESPACE=TRUE  # keep namespace after PR check for time defaulty reserverd by bonfire (optional, default: unset)
```
Comment (bsquizz) [inline cicd/_common_deploy_logic.sh:8]:
```suggestion
#RELEASE_NAMESPACE="true"  # release namespace after PR check ends (default: true)
```

File: cicd/_common_deploy_logic.sh [hunk 2]
```diff
@@ -67,7 +68,10 @@ function teardown {
     if [ ! -z "$NAMESPACE" ]; then
         set +e
         collect_k8s_artifacts
-        bonfire namespace release $NAMESPACE
+        if [ "${KEEP_NAMESPACE:+TRUE}" != "TRUE" ]; then
```
Comment (bsquizz) [inline cicd/_common_deploy_logic.sh:71]:
```suggestion
        if [ "${RELEASE_NAMESPACE:-true}" != "false" ]; then
```

**Review by bsquizz — CHANGES_REQUESTED** [review]
What if we use RELEASE_NAMESPACE env var -- by default set to true. If you do not want to release the namespace, set `RELEASE_NAMESPACE="false"` in your pr check.

**Comment by psegedy** [comment]
makes sense, I've added your changes

**Review by bsquizz — APPROVED** [review]

**Pull request merged**